### PR TITLE
Fix issue with "Add Import" code fix

### LINKED ISSUE #40857: Csharpaddimportcodefixprovider crashing due to Roslyn and nuget
**Version Used**: 16.4.2

**Steps to Reproduce**:
1. Referenced a type in my class that didn't have the appropriate "using" import statement yet
2. Try to resolve the type by importing the using through the quick actions
3. Also have a nuget.config that did not have the correct credentials to reach nuget
4. I also want add that this happens for types that are system types (like System.Collections.Generic.Dictionary) as well as solution defined types. This doesn't only happen when I try to reference types that are only available through nuget packages.

**Expected Behavior**:
Using the quick actions, the sln defined type is found and the using statement is added to the class, allowing the type to be used.

**Actual Behavior**:
Quick actions is opened and then at the top I get a "encountered an error" message for Csharpaddimportcodefixprovider with the stack trace of:

```
System.ArgumentException : Value cannot be null or empty string.
Parameter name: password
   at NuGet.Configuration.CredentialsItem..ctor(String name,String username,String password,Boolean isPasswordClearText,String validAuthenticationTypes)
   at NuGet.Configuration.CredentialsItem.Clone()
   at NuGet.Configuration.VirtualSettingSection.<>c.<Clone>b__7_0(SettingItem s)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at NuGet.Configuration.SettingsGroup`1..ctor(IReadOnlyDictionary`2 attributes,IEnumerable`1 children)
   at NuGet.Configuration.SettingSection..ctor(String name,IReadOnlyDictionary`2 attributes,IEnumerable`1 children)
   at NuGet.Configuration.VirtualSettingSection.Clone()
   at NuGet.Configuration.Settings.GetSection(String sectionName)
   at NuGet.PackageManagement.VisualStudio.VSSettings.GetSection(String sectionName)
   at NuGet.Configuration.PackageSourceProvider.ReadCredential(String sourceName,ISettings settings)
   at NuGet.Configuration.PackageSourceProvider.ReadPackageSource(SourceItem setting,Boolean isEnabled,ISettings settings)
   at NuGet.Configuration.PackageSourceProvider.LoadPackageSourceLookup(Boolean byName,ISettings settings)
   at NuGet.Configuration.PackageSourceProvider.LoadPackageSources(ISettings settings,IEnumerable`1 defaultPackageSources)
   at NuGet.Configuration.PackageSourceProvider.LoadPackageSources()
   at NuGet.VisualStudio.VsPackageSourceProvider.GetSources(Boolean includeUnOfficial,Boolean includeDisabled)
   at Microsoft.VisualStudio.LanguageServices.Packaging.PackageInstallerService.GetPackageSources()
   at async Microsoft.CodeAnalysis.AddImport.AbstractAddImportCodeFixProvider.RegisterCodeFixesAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.CodeFixService.GetCodeFixesAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
```

### PR DESCRIPTION
**Pull request by chris-schuetz**
Fixes #40857

### CODE REVIEW

File: src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs [hunk 1]
```diff
@@ -56,9 +56,19 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
+            ImmutableArray<PackageSource> packageSources;
+            try
+            {
+                // Ensure any issues with getting access to package sources will not break the Code Fix for local imports
+                // https://github.com/dotnet/roslyn/issues/40857
+                packageSources = searchNuGetPackages && symbolSearchService != null && installerService?.IsEnabled(document.Project.Id) == true
+                    ? installerService.GetPackageSources()
+                    : ImmutableArray<PackageSource>.Empty;
+            }
+            catch
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs]: definitely don't want t to just blindly catch here.  Can we at least report the issue through our watson infrastructure in order ot make sure underlying issues are found and fixxed?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs]: also, can we extract this out to a `GetPackageSources()` helper?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs]: look for `ReportWithoutCrash`

Comment (chris-dd) [reply src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs]: Thanks for your input. I found the FatalError.ReportWithoutCrash.

Also while looking into adding a GetPackageSources() helper, I found that there were already two exceptions being caught, that were related to invalid nuget.config. I undid my previous changes and moved the check for the ArgumentException as well as the ReportWithoutCrash there. The other two exceptions at that location seem to be caught for the same reason.

File: src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs [hunk 2]
```diff
@@ -108,9 +108,10 @@ public ImmutableArray<PackageSource> GetPackageSources()
                 packageSources = _packageSourceProvider.Value.GetSources(includeUnOfficial: true, includeDisabled: false)
                     .SelectAsArray(r => new PackageSource(r.Key, r.Value));
             }
-            catch (Exception ex) when (ex is InvalidDataException || ex is InvalidOperationException)
+            catch (Exception ex) when (ex is InvalidDataException || ex is InvalidOperationException || ex is ArgumentException)
             {
                 // These exceptions can happen when the nuget.config file is broken.
+                FatalError.ReportWithoutCrash(ex);
                 packageSources = ImmutableArray<PackageSource>.Empty;
             }
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs]: can you add a new catch?   the existing exceptions are known and have lots of issues already reported for them.  we don't need them rereported.  this new issue seems important so we can get data on how many times it is happening and push for teh nuget team to fix it.  thanks!

**Pull request merged**